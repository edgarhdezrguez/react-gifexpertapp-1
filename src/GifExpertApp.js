
import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

// const categories = ['One punch', 'Boku no Hero Academy', 'Dragon ball'];
    // const [categories, setCategories] = useState(['Dragon ball']);
    const [categories, setCategories] = useState(defaultCategories);
    
    // const handleAdd = () => {
    //     const newSerie = document.querySelector("#serie2Add").value;
    //     // setCategories([...categories, newSerie]);
    //     setCategories(categs => [...categs, newSerie]);
    // };
    
    return (
        <>
            <h2>GifExpertApp</h2>
            {/* <input type="text" name="addSerie" id="serie2Add"/>
            <button onClick={handleAdd}> Add </button> */}
            <AddCategory setCategories={setCategories}/>

            <hr/>
            <ol>
                {categories.map((category) => (
                <GifGrid 
                    key={category} 
                    category={category}/>
                ))}
            </ol>
        </>
    );

};

export default GifExpertApp;