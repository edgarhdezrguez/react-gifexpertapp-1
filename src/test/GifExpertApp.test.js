import React from 'react';
import { shallow, configure } from 'enzyme';
import '@testing-library/jest-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import GifExpertApp from '../GifExpertApp';


configure({ adapter: new Adapter() });
describe('Testing GifExpertApp', () => {
    
    test('should be show correctly', () => {
        
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();

    });

    test('should be show a list of categories', () => {
        
        const categories = ['Dragon ball', 'Boku no Hero'];

        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
    })
    
    




});
