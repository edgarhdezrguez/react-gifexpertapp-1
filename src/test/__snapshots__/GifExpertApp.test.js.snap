// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifExpertApp should be show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
    <GifGrid
      category="Boku no Hero"
      key="Boku no Hero"
    />
  </ol>
</Fragment>
`;

exports[`Testing GifExpertApp should be show correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
